src/main/java/org/wise/portal/service/peergroup/impl/PeerGroupServiceImpl.java

```diff
@@ -23,142 +23,69 @@
-      throws PeerGroupCreationException {
-    try {
-      List<Workgroup> workgroupsInPeriod = runDao.getWorkgroupsForRunAndPeriod(
-          workgroup.getRun().getId(), workgroup.getPeriod().getId());
-      workgroupsInPeriod.removeIf(workgroupInPeriod -> workgroupInPeriod.getMembers().size() == 0);
-      List<Workgroup> workgroupsInPeerGroup = peerGroupDao.getWorkgroupsInPeerGroup(peerGrouping,
-          workgroup.getPeriod());
-      Set<Workgroup> workgroupsNotInPeerGroup = getWorkgroupsNotInPeerGroup(workgroupsInPeriod,
-          workgroupsInPeerGroup);
-      if (isLastOnesLeftToPair(workgroupsInPeerGroup, workgroupsInPeriod,
-          peerGrouping.getMaxMembershipCount())) {
-        return workgroupsNotInPeerGroup;
-      } else {
-        return getWorkgroupsInPeerGroupUpToMaxMembership(workgroup, peerGrouping,
-            workgroupsNotInPeerGroup);
-      }
-    } catch (JSONException e) {
-      throw new PeerGroupCreationException();
-    }
-  }
-
-  private boolean isLastOnesLeftToPair(List<Workgroup> workgroupsInPeerGroup,
-      List<Workgroup> workgroupsInPeriod, int maxMembershipCount) {
-    int numWorkgroupsNotInPeerGroup = workgroupsInPeriod.size() - workgroupsInPeerGroup.size();
-    return (numWorkgroupsNotInPeerGroup - maxMembershipCount) <= 1;
-  }
-
-  private Set<Workgroup> getWorkgroupsInPeerGroupUpToMaxMembership(Workgroup workgroup,
-      PeerGrouping peerGrouping, Set<Workgroup> possibleMembers) throws JSONException {
-    Set<Workgroup> members = new HashSet<Workgroup>();
-    members.add(workgroup);
-    possibleMembers.remove(workgroup);
-    addMembersRandomly(peerGrouping, possibleMembers, members);
-    return members;
-  }
-
-  private void addMembersRandomly(PeerGrouping peerGrouping, Set<Workgroup> possibleMembers,
-      Set<Workgroup> members) {
-    List<Workgroup> possibleMembersList = new ArrayList<Workgroup>(possibleMembers);
-    Random random = new Random();
-    while (members.size() < peerGrouping.getMaxMembershipCount()) {
-      int randomInt = random.nextInt(possibleMembersList.size());
-      Workgroup randomWorkgroup = possibleMembersList.get(randomInt);
-      members.add(randomWorkgroup);
-      possibleMembersList.remove(randomWorkgroup);
-    }
-  }
-
-  private Set<Workgroup> getWorkgroupsNotInPeerGroup(List<Workgroup> workgroupsInPeriod,
-      List<Workgroup> workgroupsInPeerGroup) throws JSONException {
-    Set<Workgroup> workgroupsNotInPeerGroup = new HashSet<Workgroup>();
-    for (Workgroup workgroup : workgroupsInPeriod) {
-      if (!workgroupsInPeerGroup.contains(workgroup)) {
-        workgroupsNotInPeerGroup.add(workgroup);
-      }
+    if (peerGroupThresholdService.isThresholdSatisfied(peerGrouping, workgroup.getPeriod())) {
+      return peerGrouping.getLogic().equals("random")
+        ? randomLogicService.createPeerGroup(workgroup, peerGrouping)
+        : differentIdeasLogicService.createPeerGroup(workgroup, peerGrouping);
```

Comment:
Should we explicitly check for differentIdeas logic? If anything, maybe we should default to random instead of differentIdeas.

Reply:
Does it matter which case we test? There are only two possible logic options at this point, "random" and "differentIdeas(nodeId, componentId)" ("manual" was already handled prior to this function being called). So if it's not "random", it must be "differentIdeas". If the logic is not one of "manual", "random", or "differentIdeas", I don't think this function should be called. If we test for "differentIdeas", we have to run a Matcher (which is more expensive than an "equals" check), and "differentIdeas" occurs less often than "random", so wouldn't we want to avoid testing for it?